src/views/genres/EditGenre.vue

```diff
@@ -60,14 +60,12 @@ export default {
     fillValues() {
       this.newGenre.name = this.genre.name;
     },
-    submit() {
-      this.update({ id: this.genre.id, newGenre: this.newGenre }).then(
-        (succeeded) => {
-          if (succeeded) {
-            this.$router.push(this.$route.query.redirect || { name: "genres" });
-          }
-        }
-      );
+    async submit() {
+      const newGenre = { id: this.genre.id, newGenre: this.newGenre };
+      const succeeded = await this.update(newGenre);
```

Comment:
Why introduces the extra `const newGenre` here? I wouldn't do this; there are now three things called `newGenre` in this context.